app/src/main/java/com/github/h3lp3rs/h3lp/forum/ForumCategory.kt

```diff
@@ -23,17 +29,31 @@ enum class ForumCategory {
         fun forumOf(choice: ForumCategory): Forum {
             // The categories are the first sub-level of the forum, thus they lie after root (the
             // empty list)
-            choice.forum = choice.forum ?: FireForum(listOf(choice.name))
-            return choice.forum!!
+            root?.let {
+                return it.child(choice.name)
+            }

```

Comment:
same remark as above, I'm not sure why you use directly FireDatabase(ROOT_FORUM_DB_PATH) instead of databaseOf(FORUM) as we had beforehand since this seems like an abstraction leak. We should at least have the choice to choose the db to use in the forum if we switched from Firebase (for example after having had too many bad experiences with it :))

Reply:
My philosophy was the following:
- Databases are for, well, databases
- The forum is implemented with a database (here Firebase). It is not a database per sea, just the support. 
- Cf. previous remark
- If we want to switch from Firebase, we should be able to keep the SimpleDBForum abstract class that is Database agnostic, so I don't see the issue here.
Though I get your point, I still believe that having the public available support of the forum (databaseOf(FORUM)) accessible a bad idea.